driver-core/src/main/com/mongodb/internal/session/ServerSessionPool.java

```diff
@@ -141,16 +145,21 @@ private void endClosedSessions() {
         }).getConnection();
         try {
             connection.command("admin",
-                    new BsonDocument("endSessions", new BsonArray(closedSessionIdentifiers)), new NoOpFieldNameValidator(),
+                    new BsonDocument("endSessions", new BsonArray(identifiers)), new NoOpFieldNameValidator(),
                     ReadPreference.primaryPreferred(), new BsonDocumentCodec(), NoOpSessionContext.INSTANCE);
         } catch (MongoException e) {
             // ignore exceptions
         } finally {
-            closedSessionIdentifiers.clear();
             connection.release();
         }
     }
 
+    private void clearClosedSessionIdentifiers() {
```

Comment:
inline this and synchronize in the caller

Reply:
Done